### PR TITLE
Move websocket development server from Redwood to Zipline

### PR DESCRIPTION
**Pull request by adrw**

### CODE REVIEW

File: zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/FastCodeUpdates.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package app.cash.zipline.loader
```
Comment (swankjesse) [inline zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/FastCodeUpdates.kt:16]: supernit: copyright

(I think we might be due to get Spotless managing this?)

File: zipline-loader/src/commonTest/kotlin/app/cash/zipline/loader/FastCodeUpdatesTest.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package app.cash.zipline.loader
```
Comment (swankjesse) [inline zipline-loader/src/commonTest/kotlin/app/cash/zipline/loader/FastCodeUpdatesTest.kt:16]: Copyright!

File: zipline-loader/src/commonTest/kotlin/app/cash/zipline/loader/FastCodeUpdatesTest.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+      // close channel
+      client.closeUpdatesChannel()
+
+      // await URL flow emitting twice: once at time 0, once at 500ms on polling fallback
+      //    from socket failure
+      fastCodeFlow.take(2).toList()
+    }
+
+    // confirm >= 500 ms has elapsed
```
Comment (swankjesse) [inline zipline-loader/src/commonTest/kotlin/app/cash/zipline/loader/FastCodeUpdatesTest.kt:118]:
```suggestion
    // Confirm 500 ms has elapsed.
```

**Review by swankjesse — APPROVED** [review]

File: zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/FastCodeUpdates.kt (hunk 1 above)
Comment (JakeWharton) [reply zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/FastCodeUpdates.kt:16]: I'll do this. I just had to do it for paraphrase and will be updating redwood and friends with the new setup.

File: zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineServeTask.kt [hunk 4]
```diff
@@ -61,7 +61,6 @@ abstract class ZiplineServeTask : DefaultTask() {
       val ws = websockets(
         "/ws" bind { ws: Websocket ->
           websockets.add(ws)
-          ws.send(WsMessage(RELOAD_MESSAGE))
```
Comment (adrw) [inline zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineServeTask.kt:64]: This is a deliberate change from the current implementation in Redwood which assumes that a reload is sent initially. Instead we immediately pass through and don't wait on websocket connect.

File: zipline-loader/src/commonTest/kotlin/app/cash/zipline/loader/FastCodeUpdatesTest.kt (hunk 2 above)
Comment (adrw) [reply zipline-loader/src/commonTest/kotlin/app/cash/zipline/loader/FastCodeUpdatesTest.kt:16]: Lawyers! Be thee at ease...

**Review by yissachar — APPROVED** [review]

File: zipline-loader/src/commonTest/kotlin/app/cash/zipline/loader/FastCodeUpdatesTest.kt (hunk 2 above)
Comment (swankjesse) [reply zipline-loader/src/commonTest/kotlin/app/cash/zipline/loader/FastCodeUpdatesTest.kt:16]: It is 2023 tho

**Pull request merged**